contracts/PricerRule.sol

```diff
@@ -0,0 +1,408 @@
+
+        require(
+            payCurrencyCode != bytes3(0),
+            "Price oracle's quote currency code is null."
+        );
+
+        require(
+            baseCurrencyPriceOracles[payCurrencyCode] == address(0),
+            "Price oracle already exists."
+        );
+
+        require(
+            priceOracle.getBaseCurrencyCode() == baseCurrencyCode,
+            "Price oracle's base currency code does not match."
+        );
+
+        baseCurrencyPriceOracles[payCurrencyCode] = priceOracle;
+
+        emit PriceOracleAdded(priceOracle);
+    }
+
+    /**
+     * @notice Removes the price oracle for the specified pay currency code.
+     *
+     * @dev Function requires:
+     *          - Only organization's workers are allowed to call the function.
+     *          - Price oracle matching with the specified pay currency code
+     *            exists.
+     */
+    function removePriceOracle(
+        bytes3 _payCurrencyCode
+    )
+        external
+        onlyWorker
+    {
+        PriceOracleInterface priceOracle = baseCurrencyPriceOracles[
+            _payCurrencyCode
+        ];
+
+        require(
+            priceOracle != address(0),
+            "Price oracle to remove does not exist."
+        );
+
+        delete baseCurrencyPriceOracles[_payCurrencyCode];
+
+        emit PriceOracleRemoved(priceOracle);
+    }
+
+    /**
+     * @notice Adds an acceptance margin of the base currency price in the
+     *         specified pay currency.
+     *
+     * @dev Function requires:
+     *          - Only organization's workers are allowed to call the function.
+     *          - The specified pay currency code is not null.
+     */
+    function addAcceptanceMargin(
+        bytes3 _payCurrencyCode,
+        uint256 _acceptanceMargin
+    )
+        external
+        onlyWorker
+    {
+        require(
+            _payCurrencyCode != bytes3(0),
+            "Pay currency code is null."
+        );
+
+        baseCurrencyPriceAcceptanceMargins[_payCurrencyCode] = _acceptanceMargin;
+
+        emit AcceptanceMarginAdded(
+            _payCurrencyCode,
+            _acceptanceMargin
+        );
+    }
+
+    /**
+     * @notice Removes an acceptance margin of the base currency price in the
+     *         specified pay currency.
+     *
+     * @dev Function requires:
+     *          - Only organization's workers are allowed to call the function.
+     *          - The specified pay currency code is not null.
+     */
+    function removeAcceptanceMargin(
+        bytes3 _payCurrencyCode
+    )
+        external
+        onlyWorker
+    {
+        require(
+            _payCurrencyCode != bytes3(0),
+            "Pay currency code is null."
+        );
+
+        delete baseCurrencyPriceAcceptanceMargins[_payCurrencyCode];
+
+        emit AcceptanceMarginRemoved(_payCurrencyCode);
+    }
+
+
+    /* Private Functions */
+
+    function getBaseCurrencyPrice(
```

Comment:
The analogous function in `Pricer` is `public`—does this change indicate you find `public` visibility lacks (sufficient) value or is harmful, or does it indicate something else?

Reply:
I was thinking when we should have public/external functions in contracts.
And the simple answer is: if it's part of a contract interface.

Anything public/external could be used, hence, we should keep compatibility of that function in later versions of contract. Same goes actually to automatically generated getters. What are your thoughts?

Reply:
ok making it private

Reply:
Sounds like a vote for lacks sufficient value/is harmful, and that makes sense to me. So I am also on board for private.